### DISCUSSION TITLE
Gate with open, close, opening, closing, collision

### DISCUSSION THREAD

**Discussion by n00bster-za** [arendst/Tasmota#15545] [Q&A] [answered]
Hi. I don't use GitHub a lot so not sure if this is the right place to post this. 

I have flashed a Sonoff 1channel inching/self locking board with Tasmota 11. 

Now I am trying to get the GPIO 14 configured as a gate status. 

The question I have though is that I don't want to turn it into a switch because then it only has on and off function. The gate I am connecting it to  has, open, close,opening,closing, collision and also a power state if the main power supply is off. 

How to I configure the GPIO14 to possible receive all these status update from gate.

**Comment by barbudor** [comment, marked as answer]
A GPIO only has 2 states : ON/OFF, or HIGH/LOW or 1/0 as you wish.

So you may first explain how your garage door system provide all those states : "open, close,opening,closing, collision and also a power state if the main power supply is off."
There is already a few discussion here about garage doors that provides a status by flashing. You may want to read this one for exemple: https://github.com/arendst/Tasmota/discussions/14897

**Comment by n00bster-za** [comment]
I have read all the threads, well almost all. None provide info I need.  So a GPIO cannot read more than 2 variables coming on? its either one or the other?

**Reply by sfromis** [reply]
Unless it is the analog pin (which gpio14 is not), it can only be either on or off. Technically also floating, but you cannot detect that. Only other variation with a digital input pin is patterns of timing/pulsing/flashing. It is possible that the old firmware used a combination of simple detection combined with last actions to make assumptions about other states.

**Comment by n00bster-za** [comment]
okay so on a ESP8285 chip are there any pins that can be used as Analog?

**Reply by sfromis** [reply]
Yes, pin 6 is Tasmota gpio 17, analog. https://tasmota.github.io/docs/Pinouts/
Beware that the pin only works up to 1V, much more than that is likely to fry the chip.

First you need to work out what type of signaling it is, if it even is analog....

**Comment by n00bster-za** [comment]
how can I check the signaling. I had a look on the Itead manual there seems to be Serial Peripheral Interface (SPI/HSPI) which is GPIO 7, 8, 9, 10, 11. what can I connect to the gate to try and read the output.  

my Gate info. 
Input / Output sink currents (Max. accessory current draw)
 I/O 1-4 100mA (12/24V)

so because it is 12v I would need t step it down? Current is not that high so might survive?

**Reply by barbudor** [reply]
Not all pins are easily available on a Sonoff device

Can you please elaborate more on the Gate information ? 
Or link to a manual that explain with details the interface ?

Your questions let me believe that you are not fluent with electronics. So please provide as much details without guessing so we can guide you.

**Reply by n00bster-za** [reply]
What do you mean by not easily accessible? 

Yes I only know a little bit of electronics otherwise I would not ask on here for help. 

The gate manual only gives the info of what I gave you the I/O 4. Voltage and Amps that is it. There is nothing else that can do status output for the gate. It's a smart gate motor so I thought that it will output something that is readable with the ESP chip that this board is using.

I asked the manufacturer what is the actual output the IP gives but they said they cannot tell me it's company secret. 

But there must be someway to read what is coming out. 

Other people with the same gate is using GPIO 14 but only for open close. I was just hoping I could get more out of it possibly. 

I will see if I can attach the installer manual for the gate.

**Reply by sfromis** [reply]
"Not easily available" means going directly for one of the 8 small pins on each edge of the 5x5 mm ESP8285 chip. Possible, but many do not find that to be an attractive option. If the gate manual is "low on details", it may be up to you to explore the hardware to find out how it works.

**Comment by n00bster-za** [comment]
that is not a problem I have already flashed the unit with tasmota just by holding the pins on the chip as well did not even solder it, there is also no header pins. can you suggest. so if I solder pin 6(GPIO17) is should be able to read what the gate is pushing out for each state?

**Reply by barbudor** [reply]
Before connecting an ESP to a totally unknown signal, I would qualify better the signal.
An oscilloscope for example would be your friend or  real documentation from the manufacturer.
Until that, you have no idea of the signal levels and waveform so no idea on where you should connect it and what you could expect (apart magic smoke)

**Reply by ascillato** [reply]
@n00bster-za 

Hi,

Please, post information about your device. Post the manual, post photos of the device and the board, the wiring and also please tell us the brand and model. Without information no one will be able to help you.

**Comment by n00bster-za** [comment]
the installer manual for the gate motor at the bottom. Thanks so far for all the replies I appreciate it 

I have a 2 channel relay board.  https://ewelink.eachen.cc/product/eachen-st-dc2/

I do not yet have it wired to the Gate. I want to solder the correct GPIO first for gate status. 

I have a friend that is using the Sonoff SV and using it on the same gate motor and he is getting normal on off from the gate status output.. so I know it can work. All I wanted to try is just get that extra little bit of info as to if it's running closing or maybe more important if there was a collision. 

Actual picture of the board I have

![20220421_120736](https://user-images.githubusercontent.com/62851465/167026072-dc1e7060-4a12-44e7-800e-00734fd23a97.jpg)





[Centurion-Systems-D5-Smart-installation-manual.pdf](https://github.com/arendst/Tasmota/files/8636016/Centurion-Systems-D5-Smart-installation-manual.pdf)

**Reply by ascillato** [reply]
From the manual you post, the only information that the garage opener has is:

![image](https://user-images.githubusercontent.com/35405447/167028255-88e9b51a-933a-4061-a1bd-75cb9a4b0871.png)

So, seems that you can only:
* Trigger the open and close by sending 12V to I/O1

And read:
* If there is something blocking the infrared beam by reading 12V in the I/O3 (Seems that is 0V- something is blocking / 12V - Free)
* If it is Open or close by reading 12V in Gate Status I/O4.

It is not mentioned opening, closing and collision. Where you have read that? Maybe you need to test each I/O with a DMM to know exactly what they do?

Also, as those are 12V you can not connect them directly to the ESP (Your ESP will burn badly as it supports only 3.3V). Besides, your board don't have inputs and soldering is needed plus adaption of signals. It is not plug and play. Maybe you need a different board for this task and some work to put all together. The manual also shows a device that does what you want. Maybe using the official device would be better for you? _(I just say this because in this case, going to the DIY way might take a lot of your time if you just want to control your gate with your phone. The manufacturer already provides you with a controlling device and an app)_

**Comment by n00bster-za** [comment]
I want everything in Home assistant. The devices the manufactures give are not working with Home assistant. The app they have works only on Bluetooth not wifi.   to trigger the gate is also easy as the board as 2 relays so one will trigger full open IO1 and the other will be for pedestrian mode. IO2

I am aware that soldering is needed I have the tools to do it.  I will use Optocoupler to step down the voltages. 

I am getting a Multi meter  so will try and see what comes out. 

the gate manual says the collision is measured electronically I just assumed it was also the status output of IO4.
